src/posix/http_client_curl.cc

```diff
@@ -41,12 +44,15 @@ std::string RunCurl(const std::string& cmd) {
   FILE* pipe = ::popen(cmd.c_str(), "r");
   assert(pipe);
   char s[8 * 1024];
-  ::fgets(s, sizeof(s) / sizeof(s[0]), pipe);
+  std::string result;
+  while (nullptr != ::fgets(s, sizeof(s) / sizeof(s[0]), pipe)) {
+    result += s;
```

Comment:
I'd expect both lines to either have or not have a `'\n`' at the end.

Otherwise, the idea of taking a `- 3` doesn't look solid at all.